sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobAsyncClient.java

```diff
@@ -733,11 +734,16 @@ private <T> Mono<T> populateRequestConditionsAndContext(BlobRequestConditions re
 
                 requestConditionsFinal.setIfMatch(response.getValue().getETag());
                 Mono<T> result = downloadCall.get();
-                if (response.getValue().getMetadata().get(ENCRYPTION_DATA_KEY) != null) {
+
+                Map<String, String> caseInsensitiveMetadata = new HashMap<>();
+                for (Map.Entry<String, String> entry : response.getValue().getMetadata().entrySet()) {
+                    caseInsensitiveMetadata.put(entry.getKey().toLowerCase(Locale.ROOT), entry.getValue());
+                }
```

Comment:
If the only requirement is to get `ENCRYPTION_DATA_KEY` in a case-insensitive way I would change this design a bit. We don't need to create the case insensitive map as it's just thrown away and could do something like:

```java
static String getEncryptionDataKey(Map<String, String> metadata) {
    if (CoreUtils.isNullOrEmpty(metadata) {
        return null;
    }

    for (Map.Entry<String, String> entry : metadata.entrySet()) {
        if (entry.getKey().length() != ENCRYPTION_DATA_KEY.length()) {
            continue;
        }

        if (ENCRYPTION_DATA_KEY.regionMatches(true, 0, entry.getKey(), 0, ENCRYPTION_DATA_KEY.length()) {
            return entry.getData();
        }
    }
}
```

This removes all the string lowercasing, map insertion, and then the final map get with a single pass over all entries in the map filtering out those which don't match ENCRYPTION_DATA_KEY's length and returns when the key matches case insensitively.